pages/glossary/programatically-determined-link-context.md

```diff
@@ -0,0 +1,19 @@
+--
+title: Programmatically Determined Link Context
```

Comment:
I think you're making an assumption here that HTML is valid. Let's take something like this:

```html
<p>
  <h1>ACT Rules Community Group</h1>
  <p><a href="...">learn more</a></p>
</p>
```

I would argue that this should fail the SC, but according to this definition, the heading text is part of the context, and so it passes. I don't know if that's a bad thing necessarily. This doesn't cause false positives, but it doesn't tell the whole story either. I'm OK ignoring this edge case, but if we do, I think there needs to be an assumption.

Reply:
I agree that this should fail the SC. I've updated the definition to look only at the nearest ancestor of the link element, so the heading is no longer part of the context.

Reply:
There's a lot of repetition in this new text. You've also only solved the case of the element being the same, But similarly to the previous example, I could swap out the top level `p` for another element in your list, and you'd have the same problem again:

```html
<li>
  <h1>ACT Rules Community Group</h1>
  <p><a href="...">learn more</a></p>
</li>
```

Additionally, this no longer allow nested list items to be part of context, like you have in passed example 5. This would now fail the rule:


```html
<ul>
  <li> Ulysses <ul>
    <li><a href="...">HTML</a></li>
    <li><a href="...">EPUB</a></li>
    <li><a href="...">Plain text</a></li>
  </ul></li>
</ul>
```

Reply:
Regarding the first example, I now would say that this counts as context for this SC (i.e., the text from the heading is in the same listitem). It is my interpretation of the understanding document that context comes from the nearest ancestor (otherwise it should have been written "includes text that is in the same paragraph**s**, list**s**, or table cell**s**...")

This interpretation does have the side effect of now failing passed example 5 (nicely spotted). I've updated it to use the aria-describedby attribute to provide the required context.

I don't really know what to do about the text repetition. I wasn't very happy with it, but this was the most succint I managed to be. Any suggestions to improve it, @WilcoFiers?

Reply:
Thanks for the suggestion. In it you are listing elements and not relationships, so I've updated it to list relationships.